src/sentry/static/sentry/app/views/eventsV2/landing.tsx

```diff
@@ -157,21 +152,25 @@ class DiscoverLanding extends AsyncComponent<Props, State> {
     }
   }
 
-  getDocumentTitle = (eventView: EventView): string => {
-    return generateTitle({
-      eventView,
-    });
+  handleQueryChange = () => {
+    this.fetchData({reloading: true});
   };
 
   handleClick = () => {
     localStorage.setItem(BANNER_DISMISSED_KEY, 'true');
     this.setState({isBannerHidden: true});
   };
 
-  // When a query is saved or deleted we need to re-fetch the
-  // saved query list as we don't use a reflux store.
-  handleQueryChange = () => {
-    this.fetchData({reloading: true});
+  handleSearchQuery = (searchQuery: string) => {
+    const {location} = this.props;
+    ReactRouter.browserHistory.push({
+      pathname: location.pathname,
+      query: {
+        ...location.query,
+        cursor: undefined,
+        query: String(searchQuery).trim() || undefined,
+      },
+    });
   };
 
   renderBanner() {
```

Comment:
@markstory the "Build a new query" CTA button on the banner will need to have its route updated.